composer/src/main/kotlin/com/gojuno/composer/Instrumentation.kt

```diff
@@ -133,27 +137,29 @@ fun Observable<InstrumentationEntry>.asTests(): Observable<InstrumentationTest>
                             val second = entries
                                     .subList(index + 1, entries.size)
                                     .firstOrNull {
-                                        first.clazz == it.clazz
-                                                &&
-                                                first.test == it.test
-                                                &&
-                                                first.current == it.current
-                                                &&
-                                                first.statusCode != it.statusCode
+                                        first.clazz == it.clazz &&
+                                        first.test == it.test &&
+                                        first.current == it.current &&
+                                        first.statusCode != it.statusCode
                                     }
 
                             if (second == null) null else first to second
                         }
                         .filterNotNull()
                         .map { (first, second) ->
                             InstrumentationTest(
+                                    index = first.current,
```

Comment:
Printing test index and total test count sharded to particular device really helps.